Source/loadsave.cpp

```diff
@@ -128,25 +128,20 @@ class LoadHelper {
 	{
 		return next<uint32_t>() != 0;
 	}
-
-	~LoadHelper()
-	{
-		mem_free_dbg(m_buffer);
-	}
 };
 
 class SaveHelper {
 	const char *m_szFileName;
-	uint8_t *m_buffer;
+	std::unique_ptr<uint8_t[]> m_buffer;
 	uint32_t m_cur = 0;
 	uint32_t m_capacity;
 
 public:
 	SaveHelper(const char *szFileName, size_t bufferLen)
+		: m_szFileName(szFileName)
+		, m_buffer(std::make_unique<uint8_t[]>(codec_get_encoded_len(bufferLen)))
+		, m_capacity(bufferLen)
 	{
-		m_szFileName = szFileName;
-		m_capacity = bufferLen;
-		m_buffer = DiabloAllocPtr(codec_get_encoded_len(m_capacity));

```

Comment:
What is the benifit of doing the assignment in the signature instead of the body?

Reply:
In terms of practicality, none.

Reply:
typical C++ to have an alternative syntax for things